STORE variants: SINTER,SUNION,SDIFF,ZUNION use setKey instead of dbDelete+dbAdd

yangbodong22011 opened the pull request:
solves #7484

soloestoy commented on src/t_set.c:
```diff
@@ -895,16 +895,16 @@ void sinterGenericCommand(client *c, robj **setkeys,
         if (setTypeSize(dstset) > 0) {
-            dbAdd(c->db,dstkey,dstset);
+            setKey(c,c->db,dstkey,dstset);
+            decrRefCount(dstset);
```
these two `decrRefCount` can merge and move out of if ... else scope.

yangbodong22011 replied:
Thanks, fixed.

oranagra commented:
`signalModifiedKey` can now be moved into the delete `else`, since `setKey` already handles that.

p.s. it also adds `removeExpire` which was already handled by the old code from the fact the key was already deleted before dbAdd.

yangbodong22011 commented:
> `signalModifiedKey` can now be moved into the delete `else`, since `setKey` already handles that.

Thanks, fixed.

I also changed `server.dirty++` when `dbDelete` success.

oranagra commented:
thanks a lot. seems good.

soloestoy commented:
>signalModifiedKey can now be moved into the delete else, since setKey already handles that.

@oranagra you remind me about this PR #5783
